homeassistant/helpers/config_validation.py

```diff
@@ -430,6 +432,22 @@ def ensure_list_csv(value: Any) -> Sequence:
     return ensure_list(value)
 
 
+def deprecated(key):
+    """Log key as deprecated."""
+    module_name = inspect.getmodule(inspect.stack()[1][0]).__name__
+
+    def validator(config):
+        """Check if key is in config and log warning."""
+        if key in config:
+            logging.getLogger(module_name).warning(
+                "The '%s' option is deprecated, please remove it from your "
+                "configuration.", key)
```

Comment:
You could include the config dict in the log message, after "configuration". But that might look messy if it has lots of items

Reply:
I added the value of the deprecated option now.